heffte/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+find_package(Heffte 2.2 PATHS "${HEFFTE_DIR}" REQUIRED)
```

Comment:
To be consistent with other packages, can you please move the `find_package` command in the top-level CMake file?

Reply:
What I see that `heFFTe` is enabled by default and `MAGMA` is disabled by default. Maybe we need logic that enables examples based on a mix of options that were enabled in top `CMakeLists.txt`.

Reply:
The if-statement guarding this is in the main CMakeLists.txt and by default heFFTe is not enabled and the entire subdirectory will not be included.

But @luszczek has a good point, packages are set to on/off seemingly at random. We should have some better criteria.

Reply:
Moved the `find_package()`

Reply:
merged to `testing-v0.4.0` branch

Reply:
> The if-statement guarding this is in the main CMakeLists.txt and by default heFFTe is not enabled and the entire subdirectory will not be included.
> 
> But @luszczek has a good point, packages are set to on/off seemingly at random. We should have some better criteria.

I thought the logic here is that this example can work both with and without MAGMA and that dependency is handled through heFFTe, so there's no need for any explicit logic regarding MAGMA in this example. Is that not correct?

Reply:
I don't think this was in regard the heFFTe-MAGAM integration but a statement regarding all packages. Right now, some packages are enabled by default and others not, seemingly a bit at random.

Reply:
Okay, I see. I guess the decision about the default state of a package, enabled or disabled, was made by the person that added the specific package. So, yes, it is a bit random, I guess.